PyDSS/dssInstance.py

```diff
@@ -420,31 +460,81 @@ def RunSimulation(self, project, scenario, MC_scenario_number=None):
         try:
             step = 0
             while step < Steps:
-                self.RunStep(step)
-
+                pydss_has_converged = True
+                opendss_has_converged = True
+                within_range = self._simulation_range.is_within_range(self._dssSolver.GetDateTime())
+                if within_range:
+                    with Timer(self._stats, "RunStep"):
+                        pydss_has_converged = self.RunStep(step)
+                        opendss_has_converged = dss.Solution.Converged()
+                        if not opendss_has_converged:
+                            self._Logger.info("OpenDSS did not converge at step=%s pydss_converged=%s",
+                                              step, pydss_has_converged)
+                has_converged = pydss_has_converged and opendss_has_converged
                 if step == 0 and self.ResultContainer is not None:
                     size = make_human_readable_size(self.ResultContainer.max_num_bytes())
                     self._Logger.info('Storage requirement estimation: %s, estimated based on first time step run.', size)
-
-                for postprocessor in postprocessors:
-                    step = postprocessor.run(step, Steps)
+                if postprocessors and within_range:
+                    step, has_converged = self._RunPostProcessors(step, Steps, postprocessors)
                 if self._increment_flag:
                     step += 1
 
+                # NOTE for review by Aadil:
+                # I moved the next two code blocks here because UpdateResults needs to happen
+                # after the postprocessors.
+                # the Helics update needs to happen after that.
```

Comment:
Note this @AadilLatif . I'll remove the comment after receiving your acknowledgement.